app/controllers/api/assessments_controller.rb

```
@@ -59,21 +59,26 @@ def show
     if assessment.summative? && !review_or_edit
       if user_assessment && @lti_launch
 
-        if user_assessment.lti_role == 'student' && user_assessment.attempts >= assessment_settings.allowed_attempts
-          render :json => {:error => "Too many attempts."}, status: :unauthorized
-          return
-        end
+        assessment_results_with_lti_launch = AssessmentResult.where(lti_launch_id: @lti_launch.id)
+        if assessment_results_with_lti_launch.count == 0
+          # only create new assessment result and update assessment attempts if this is a unique lti launch
+
+          if user_assessment.lti_role == 'student' && user_assessment.attempts >= assessment_settings.allowed_attempts
+            render :json => {:error => "Too many attempts."}, status: :unauthorized
+            return
+          end
 
-        @result = assessment.assessment_results.build
-        @result.user_assessment = user_assessment
-        @result.lti_launch = @lti_launch
-        @result.external_user_id = @lti_launch.lti_user_id if @lti_launch
-        @result.attempt = user_assessment.attempts || 0
-        @result.user = current_user
-        @result.session_status = AssessmentResult::STATUS_PENDING_SUBMISSION
-        @result.save!
+          @result = assessment.assessment_results.build
+          @result.user_assessment = user_assessment
+          @result.lti_launch = @lti_launch
+          @result.external_user_id = @lti_launch.lti_user_id if @lti_launch
+          @result.attempt = user_assessment.attempts || 0
+          @result.user = current_user
+          @result.session_status = AssessmentResult::STATUS_PENDING_SUBMISSION
+          @result.save!
 
-        user_assessment.increment_attempts!
+          user_assessment.increment_attempts!
+        end
```

If we don't have an else at this point, does that mean the user might get a free assessment if the error conditions _don't_ exist?

---

I think they would get a 200 response and the XML, but I don't think it'll get processed correctly at submission time. Not quite sure. 
I think you should put an else with a generic error message and unauthorized response. That might even help trace down why the problem is happening in the first place if a student ends up seeing it. I'd also throw an error log in that situation too with anything you think would help you figure out how it's happening. Even just a "User id triggered the extra quiz attempt thing". Then you can watch for it in the logs and find all the requests for that user before and after that line and see if you can figure out where they navigated.

Something like:

```
render :json => {:error => "Something went wrong. Please try again.."}, status: :unauthorized
return
```

---

Cool, I'll get that done first thing tomorrow.  Thank you greatly, @bracken  for taking a look!